proposals/NativeInt.md

```diff
@@ -0,0 +1,326 @@
+| `byte` | `nuint` | ImplicitNumeric | `conv.u` |
+| `short` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `ushort` | `nuint` | ImplicitNumeric | `conv.u` |
+| `int` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `uint` | `nuint` | ImplicitNumeric | `conv.u` |
+| `long` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `ulong` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `char` | `nuint` | ImplicitNumeric | `conv.u` |
+| `float` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `double` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `decimal` | `nuint` | ExplicitNumeric | `ulong UIntPtr.op_Explicit(decimal) UIntPtr UIntPtr.op_Explicit(ulong)` |
+| `IntPtr` | `nuint` | None | |
+| `UIntPtr` | `nuint` | Identity | |
+
+| Operand | Target | Conversion | IL |
+|:---:|:---:|:---:|:---:|
+| `nint` | `object` | Boxing | `box` |
+| `nint` | `string` | None | |
+| `nint` | `void*` | PointerToVoid | `conv.i` |
+| `nint` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `nint` | `sbyte` | ExplicitNumeric | `conv.i1` / `conv.ovf.i1` |
+| `nint` | `byte` | ExplicitNumeric | `conv.u1` / `conv.ovf.u1` |
+| `nint` | `short` | ExplicitNumeric | `conv.i2` / `conv.ovf.i2` |
+| `nint` | `ushort` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2` |
+| `nint` | `int` | ExplicitNumeric | `conv.i4` / `conv.ovf.i4` |
+| `nint` | `uint` | ExplicitNumeric | `conv.u4` / `conv.ovf.u4` |
+| `nint` | `long` | ImplicitNumeric | `conv.i8` / `conv.ovf.i8` |
+| `nint` | `ulong` | ExplicitNumeric | `conv.i8` / `conv.ovf.i8` |
+| `nint` | `char` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2` |
+| `nint` | `float` | ImplicitNumeric | `conv.r4` |
+| `nint` | `double` | ImplicitNumeric | `conv.r8` |
+| `nint` | `decimal` | ExplicitNumeric | `long decimal.op_Explicit(decimal) IntPtr IntPtr.op_Explicit(long)` |
+| `nint` | `IntPtr` | Identity | |
+| `nint` | `UIntPtr` | None | |
+| `nuint` | `object` | Boxing | `box` |
+| `nuint` | `string` | None | |
+| `nuint` | `void*` | PointerToVoid | `conv.u` |
+| `nuint` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
+| `nuint` | `sbyte` | ExplicitNumeric | `conv.i1` / `conv.ovf.i1` |
+| `nuint` | `byte` | ExplicitNumeric | `conv.u1` / `conv.ovf.u1` |
+| `nuint` | `short` | ExplicitNumeric | `conv.i2` / `conv.ovf.i2` |
+| `nuint` | `ushort` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2` |
+| `nuint` | `int` | ExplicitNumeric | `conv.i4` / `conv.ovf.i4` |
+| `nuint` | `uint` | ExplicitNumeric | `conv.u4` / `conv.ovf.u4` |
+| `nuint` | `long` | ExplicitNumeric | `conv.i8` / `conv.ovf.i8` |
+| `nuint` | `ulong` | ImplicitNumeric | `conv.u8` / `conv.ovf.u8` |
+| `nuint` | `char` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2.un` |
+| `nuint` | `float` | ImplicitNumeric | `conv.r.un conv.r4` |
+| `nuint` | `double` | ImplicitNumeric | `conv.r.un conv.r8` |
+| `nuint` | `decimal` | ExplicitNumeric | `ulong decimal.op_Explicit(decimal) UIntPtr UIntPtr.op_Explicit(ulong)` |
```

Comment:
>ExplicitNumeric [](start = 24, length = 15)

Since the conversion from `ulong` to `decimal` is an implicit conversion, this should be too.

Reply:
https://github.com/dotnet/csharplang/pull/2833/files#r386614583